packages/hashAlgo/hashAlgo.js

```diff
@@ -0,0 +1,118 @@
+const crypto = require('crypto')
+const EventEmitter = require('events')
+const { size, isEmpty } = require('lodash')
+
+const preEvents = [
+  {
+    callId: 1,
+    userId: 4545451,
+    method: 'foo.bar1',
+    params: {},
+  },
+  {
+    callId: 2,
+    userId: 45454512,
+    method: 'foo.bar2',
+    params: {},
+  },
+  {
+    callId: 3,
+    userId: 4545453,
+    method: 'foo.bar3',
+    params: {},
+  },
+]
+
+const postEvents = [
+  {
+    callId: 3,
+    result: '',
+  },
+  {
+    callId: 1,
+    result: '',
+  },
+  {
+    callId: 2,
+    result: '',
+  },
+]
+
+// IMHO, it will be better to use a HMAC instead of a simple hash.
+const createHash = data =>
+  crypto
+    .createHash('sha256')
+    .update(JSON.stringify(data))
+    .digest('hex')
```

Comment:
What if one day we decide to switch to another algorithm?

Reply:
To handle this case, i should add the algorithm entry in the logs, then we only need update hashes once on changing the algorithm.

Reply:
You could use this format: https://github.com/vatesfr/xen-orchestra/blob/f4fb0a1c791a3191b447846a99a74cb1a91c8b1c/%40xen-orchestra/fs/src/checksum.js#L9-L16